src/partition.ml

```diff
@@ -0,0 +1,94 @@
+open Encoding
+module H_contents = Irmin.Hash.Typed (Hash) (Contents)
+
+type entry = { name : string; kind : [ `Node | `Contents ]; hash : Hash.t }
+type enc_entry = { eencoding : string; entry : entry }
```

Comment:
maybe define `type 'a encoding = { encoding: string; v: 'a }` instead of duplicating this everywhere.